### DISCUSSION TITLE
NES Worker

### DISCUSSION THREAD

**Discussion by zeuchste** [nebulastream/nebulastream#5] [Ideas]
Open points:

Resolved points:
- [x] Improve start and shutdown of queries, especially testing them, #36 
- [x] Refactor and remove code #35 and #37
- [x] Improve Error Handling, see discussion https://github.com/nebulastream/nebulastream-public/discussions/32
- [x] Create a Common interface #34

**Comment by zeuchste** [comment]
Overall this component is nothing in itself the real stuff for the engine is in the runtime so we should maybe move this over here. Mainly this is just a grpc client and the mobility part, and the starter

If we really say the worker is for query maintenance, then we have to pull all that part from nes-runtime here.

---
Stuff to remove:

- [ ] LocationProvider and all the features for that as it is not stable
- [ ] Everything under mobility

---
Stuff to refactor:
- [ ] CallData as nobody really understands what does
- [ ] WorkerHealthCheckService as this should not be mandatory anymore
- [ ] /tests why there are unique RPC client tests here

---

**Reply by PhilippGrulich** [reply]
Maybe it makes sense to look it as the following:

Engine -> A C++ lib that (1) receives query plans, (2) compiles them, and (3) executes them.

Worker -> something that communicates with the outer world and uses the engine. 

As the result the engine should not know about the coordinator.

**Reply by zeuchste** [reply]
this sounds good to me
